org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/utils/ClientLogHandler.java

```diff
@@ -0,0 +1,58 @@
+package org.eclipse.lsp4xml.utils;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.logging.Handler;
+import java.util.logging.LogRecord;
+import java.util.logging.SimpleFormatter;
+
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+import org.eclipse.lsp4j.services.LanguageClient;
+
+/**
+ * LogHandler
+ */
+public class ClientLogHandler extends Handler{
+  private LanguageClient languageClient;
+
+  public ClientLogHandler(LanguageClient languageClient) {
+    this.languageClient = languageClient;
+  }
+
+  @Override
+  public void publish(LogRecord record) {
+    if(languageClient == null) {
+      return;
+    }
+    StringBuilder sb = new StringBuilder();
+    sb.append(record.getMillis())
+    .append(" - ")
+    .append(record.getSourceClassName())
+    .append("#")
+    .append(record.getSourceMethodName())
+    .append(" - ")
+    .append(record.getMessage());
+    if (record.getThrown() != null) {
+      StringWriter sw = new StringWriter();
+      PrintWriter pw = new PrintWriter(sw);
+      pw.println();
+      record.getThrown().printStackTrace(pw);
+      pw.close();
+      String throwable = sw.toString();
+      sb.append(throwable);
+    }
+    MessageParams mp = new MessageParams(MessageType.Info, sb.toString());
```

Comment:
messagetype should be mapped to the record level

Reply:
It's not possible, I'll get it working another way.